modules/data-entry/src/main/java/ca/sickkids/ccm/lfs/QueryBuilder.java

```diff
@@ -153,7 +159,42 @@ private String fullTextEscape(String input)
         }
 
         // Escape sequence taken from https://jackrabbit.apache.org/archive/wiki/JCR/EncodingAndEscaping_115513396.html
-        return input.replaceAll("([\\Q+-&|!(){}[]^\"~*?:\\/\\E])", "\\\\$1").replaceAll("'", "''");
+        return input.replaceAll("([\\Q+-&|!(){}[]^\"~*?:\\_%/\\E])", "\\\\$1").replaceAll("'", "''");
+    }
+
+    /**
+     * Gets the question for a given answer JCR Resource.
+     *
+     * @param res the JCR Resource corresponding to an answer
+     * @return the question string corresponding to the passed answer
+     */
+    private String getQuestion(Resource res) throws ItemNotFoundException, RepositoryException
+    {
+        String questionNodeId = res.getValueMap().get("question", "");
+        if (!"".equals(questionNodeId)) {
+            Node questionNode = res.adaptTo(Node.class).getSession().getNodeByIdentifier(questionNodeId);
+            return questionNode.getProperty("text").getValue().toString();
+        }
```

Comment:
It is much easier in this case to just access the node directly, rather than obtaining the session and obtaining the node by ID.
```suggestion
        Node questionNode = res.adaptTo(Node.class).getProperty("question").getNode();
        if (questionNode != null) {
            return questionNode.getProperty("text").getString();
        }
```

Reply:
I have addressed this issue, and other robustness fixes, in commit `61bac50`. :heavy_check_mark: